crates/lgn-api-codegen/src/rust/templates/models.rs.jinja

```diff
@@ -85,7 +85,22 @@ use lgn_online::codegen::Error;
                     {%- endif -%}
                 {% endfor %}
             }
-        {% when t %}
+        {% when Type::StructMap with { fields, map } %}
```

Comment:
Is there a reason why we can't just add the additional properties in the existing `Type::Struct` variant and handle it conditionally ?

I feel like both `Struct` and `StructMap` have a lot in common, and those shows in all the templates.